File: C#/Interface/TransformerInterface.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XmlBinder.Interface
{
	public interface TransformerInterface
	{
		/// <summary>
		/// Function to convert a string stored in file to an object.
		/// </summary>
		/// <param name="_value">The string read in the file.</param>
		/// <returns>The object converted from the string.</returns>
		object read(string _value);

		/// <summary>
		/// Function to convert an object to string to store it to a file.
		/// </summary>
		/// <param name="_value">The object to convert.</param>
		/// <returns>The string which is the converted representation of the object.</returns>
		string write(object _value);
	}
}

File: Java/src/storm/xmlbinder/transformer/TransformerInterface.java
package storm.xmlbinder.transformer;

/**
 * This interface describe the needed function for a transformer.
 * 
 * @author Storm <j-storm@hotmail.fr>
 */
public interface TransformerInterface 
{
	/**
	 * Method to read a String value and transform it into the desired object.
	 * @param _value : the value to convert.
	 * @return the converted value.
	 */
	public Object read(String _value);
	/**
	 * Method to write an object to a string.
	 * @param _value : the value to convert to a string.
	 * @return the converted string.
	 */
	public String write(Object _value);
}
